pkg/datapath/linux/route/reconciler/reconciler.go

```diff
@@ -0,0 +1,418 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package reconciler
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"iter"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/statedb"
+	"github.com/cilium/statedb/reconciler"
+	"github.com/vishvananda/netlink"
+	"go4.org/netipx"
+
+	"github.com/cilium/cilium/pkg/datapath/linux/safenetlink"
+	"github.com/cilium/cilium/pkg/datapath/tables"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/time"
+	"github.com/cilium/cilium/pkg/wal"
+)
+
+func registerReconciler(
+	params reconciler.Params,
+	tbl statedb.RWTable[*DesiredRoute],
+	devices statedb.Table[*tables.Device],
+	log *slog.Logger,
+	config *option.DaemonConfig,
+) (reconciler.Reconciler[*DesiredRoute], error) {
+	ops := newOps(params.Lifecycle, params.DB, tbl, devices, log, config)
+	return reconciler.Register(
+		params,
+		tbl,
+		(*DesiredRoute).Clone,
+		(*DesiredRoute).SetStatus,
+		(*DesiredRoute).GetStatus,
+		ops,
+		ops,
+		reconciler.WithPruning(30*time.Minute),
```

Comment:
Make this configurable?

Reply:
I mean we can, but do you actually expect anyone to configure the interval? 

Pruning always happens on startup, as soon as the initializers finish. This initial run should cleanup any stray routes. After that, its mostly a double-check mechanism in case the incremental update mechanism has a flaw. We also happen to do WAL compaction then, but if WAL growth in 30 minutes becomes an issue, then it would likely need code changes anyway.